### ISSUE TITLE
Better handling for folders containing mixed image and non-image files

### ISSUE THREAD

**Issue by GenevieveBuckley** [napari/napari#4805]
## 🚀 Feature
Better handling for folders containing mixed image and non-image files.

## Motivation

Something I find frustrating is when I drag & drop a folder onto napari and that folder contains not just image files, but also a text or csv file (often containing experiment metadata or notes about the sample). This causes an error in napari, not immediately but instead when dask attempts to fetch and read the .txt/.csv file. This makes napari not very useable, since I can't switch to 3D mode and view the data.

## Pitch

I would like to be able to drag & drop a folder full of image files plus some random .txt/.csv/etc. non-image files, and have napari be able to display the image data and switch to 3d mode without any problems.

## Alternatives

Option 1: We could try and filter out filenames that do not match reader plugins eagerly, rather than wait for dask to attempt to fetch data from the file.
Option 2: We could better handle errors as they occur, in a way that doesn't compromise the display or the image data.

## Additional context

GOOD: Here is the 3D view when I drag and drop a folder full of *only* image files. (Ignore the fact the image looks weird, I just copied a whole bunch of identical 2D image slices of the scikit-image cameraman to make demo data)
![napari-3d-view-with-only-image-files](https://user-images.githubusercontent.com/30920819/178697575-d3130f56-5deb-4413-b81f-537cb7d03b55.png)

BAD: Here is the 3D view when I drag and drop a folder containing images files *plus* some non-image .txt or .csv files. One 2D slice is visible, but the image volume is not displayed correctly.
![napari-3dview-with-non-image-files](https://user-images.githubusercontent.com/30920819/178697949-8e2b81a4-aa21-4a41-9660-c0e6685b3a37.png)

**Comment by DragaDoncila** [comment]
I think we should silently ignore non-image files. Currently iirc the logic for grabbing images from folders just takes the first path it finds and assumes the rest of them are all the same. Maybe we should be a bit cleverer there and check for things we *can* read? It's just not clear how we would handle the case where there's multiple types of images we can read in there...

**Comment by Czaki** [comment]
But how to know that file is an image?

**Comment by DragaDoncila** [comment]
@Czaki yeah that's the trick, but I think we could quasi-achieve this the way we achieve any reading - check whether any reader plugin would claim the file

**Comment by Czaki** [comment]
So maybe the fix may work if a user has only `npe2` readers?

**Comment by DragaDoncila** [comment]
For legacy plugins we could run their `napari_get_reader`, which is what we do now. Although very soon there will be no more legacy plugins and we should have filename patterns for all

**Comment by GenevieveBuckley** [comment]
> I think we should silently ignore non-image files. Currently iirc the logic for grabbing images from folders just takes the first path it finds and assumes the rest of them are all the same. Maybe we should be a bit cleverer there and check for things we _can_ read? It's just not clear how we would handle the case where there's multiple types of images we can read in there...

Oh, so I've been lucky so far in that alphabetical ordering has listed my image files first? That is absolutely not guaranteed.

I feel like an acceptable compromise would be to examine the string/path filename extension. ...Although csv does appear to be something that napari tries to open but can't? (Perhaps this was just intended for reloading shapes/points data, and causes difficulty in other contexts)

**Comment by Czaki** [comment]
Yes. Points are stored in csv. 

Other idea: `.napariignore` file?
